languages/javalang/builder-renderer/commercetools-java-client/src/main/java/client/ApiHttpRequest.java

```diff
@@ -42,6 +44,13 @@ public void setHeaders(ApiHttpHeaders headers) {
         this.headers = headers;
     }
 
+    public void addHeader(String key,String value) {
+        if(this.headers == null){
+            this.headers = new ApiHttpHeaders();
```

Comment:
Here we should at least document that this method isn't thread safe. Or we can always initializes the headers and then this code is thread safe too.

Reply:
For this one i would mention that this method is not thread safe.